### PR TITLE
chore: Fix CI

### PR DESCRIPTION
**Pull request by alamb**
# Rationale

CI appears to be failing on main: https://app.circleci.com/pipelines/github/influxdata/pbjson?branch=main due to some bitrot

* cargo_audit: https://app.circleci.com/pipelines/github/influxdata/pbjson/274/workflows/b620d772-632d-4dd8-bc83-cd9a0e01a30b/jobs/1433
* lint: https://app.circleci.com/pipelines/github/influxdata/pbjson/274/workflows/b620d772-632d-4dd8-bc83-cd9a0e01a30b/jobs/1431
* vendor: https://app.circleci.com/pipelines/github/influxdata/pbjson/274/workflows/b620d772-632d-4dd8-bc83-cd9a0e01a30b/jobs/1432

# Changes
Let's fix those failures

### CODE REVIEW

File: pbjson-build/src/generator/message.rs [hunk 1]
```diff
@@ -380,6 +380,11 @@ fn write_serialize_scalar_variable<W: Write>(
                 "{}#[allow(clippy::needless_borrow)]",
                 Indent(indent)
             )?;
+            writeln!(
+                writer,
+                "{}#[allow(clippy::needless_borrows_for_generic_args)]",
```
Comment (alamb) [inline pbjson-build/src/generator/message.rs:385]: apparently the new version of clippy flags some of this code as "needless borrows" but under a different lint (`needless_borrow` previously and now `needless_borrows_for_generic_args`)

File: .circleci/config.yml [hunk 2]
```diff
@@ -150,8 +150,8 @@ jobs:
     steps:
       - checkout
       - run:
-          name: Install protobuf and git
-          command: pacman -Sy --noconfirm protobuf git && protoc --version
+          name: Install protobuf and git and dependencies
+          command: pacman -Sy --noconfirm protobuf git gcc-libs && protoc --version
```
Comment (alamb) [inline .circleci/config.yml:154]: This is the fix for https://app.circleci.com/pipelines/github/influxdata/pbjson/274/workflows/b620d772-632d-4dd8-bc83-cd9a0e01a30b/jobs/1432

It appears the new version of protoc needs its libraries updated as well

File: pbjson-types/src/timestamp.rs [hunk 3]
```diff
@@ -8,14 +8,13 @@ impl TryFrom<Timestamp> for DateTime<Utc> {
     fn try_from(value: Timestamp) -> Result<Self, Self::Error> {
         let Timestamp { seconds, nanos } = value;
 
-        let dt = NaiveDateTime::from_timestamp_opt(
+        Self::from_timestamp(
```
Comment (alamb) [inline pbjson-types/src/timestamp.rs:11]: This is needed to get clippy passing https://app.circleci.com/pipelines/github/influxdata/pbjson/274/workflows/b620d772-632d-4dd8-bc83-cd9a0e01a30b/jobs/1431

Newer versions of Chrono have deprecated several APIs

File: rust-toolchain.toml [hunk 4]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.72"
+channel = "1.74"
```
Comment (alamb) [inline rust-toolchain.toml:2]: Needed to fix cargo audit: https://app.circleci.com/pipelines/github/influxdata/pbjson/274/workflows/b620d772-632d-4dd8-bc83-cd9a0e01a30b/jobs/1433

**Comment by alamb** [comment]
@crepererum  or @jeffreyssmith2nd  do you happen to have a few moments to review a PR to get CI passing again?

**Review by wiedld — APPROVED** [review]

**Review by crepererum — APPROVED** [review]
thanks

**Comment by alamb** [comment]
Thanks @wiedld  and @crepererum

**Pull request merged**